File: a5/backup2/other/q2Philosopher.java
public class q2Philosopher extends Thread {
	
	private int id, numberOfNoodles;
	private q2table table;
	private q2Printer printer;
    
    public enum States {
		Thinking( 'T' ), Hungry( 'H' ), Eating( 'E' ), Waiting( 'W' ), Finished( 'F' );
		public final char val;
		States( char v ) { val = v; }
    } // Philosopher.states
    
    private void doYield(int amount){
   		// System.out.println("id: " + id + " amount: " + amount);

		for(int i=0; i<amount; i++){
			yield();
			   		try { sleep(1);
			}catch(InterruptedException ex){
				System.out.println("ERROR " + ex);
			}
			
			//have to remove this
		//	System.out.println("id: " + id + " i: " + i);
		}
	}
		
	public q2Philosopher(int id, int noodles, q2table table, q2Printer prt){
	//System.out.println("id: " + id + " started");
		this.id = id;
		numberOfNoodles = noodles;
		this.table = table;
		printer = prt;	
	}	
	
	public void run(){
		//System.out.println("id: " + id + " running");
		while(true){
			printer.print(id, States.Hungry);
			
			doYield(q2PRNG.getRand(0,4));
			
			table.pickup(id);
			
			int eatAmount = (q2PRNG.getRand(1,5));
			
			if(eatAmount > numberOfNoodles)
				eatAmount = numberOfNoodles;
			
			numberOfNoodles -= eatAmount;
			
			printer.print(id, States.Eating, eatAmount, numberOfNoodles);
			
			if(numberOfNoodles == 0) {
				table.putdown(id);
				break;
			}
					
			doYield(q2PRNG.getRand(0,4));		

			table.putdown(id);
			
			printer.print(id, States.Thinking);
			
			doYield(q2PRNG.getRand(0,19));
		}
		
		printer.print(id, States.Finished);
	}



}

File: a5/backup/other2/q2Philosopher.cc
#include <uC++.h>
#include "q2Philosopher.h"
//#include "q2table.h"
#include "q2Printer.h"
#include "q2PRNG.h"

extern PRNG myPrng;

void Philosopher::main(){
	while(true){
	
		printer->print(id, Philosopher::Hungry);
		
		yield(myPrng(4));
		
		table->pickup(id);
		
		unsigned int eatAmount = myPrng(1,5);
		
		if(eatAmount > numberOfNoodles)
			eatAmount = numberOfNoodles;
		
		assert(eatAmount < 6);
		numberOfNoodles -= eatAmount;
		
		printer->print(id, Philosopher::Eating, eatAmount, numberOfNoodles);
		
		if(numberOfNoodles == 0) {
			table->putdown(id);
			break;
		}
				
		yield(myPrng(4));
		

		table->putdown(id);
		
		//if(numberOfNoodles == 0)
			//break;
			
		printer->print(id, Philosopher::Thinking);
		
		yield(myPrng(19));
		
	}
	
	printer->print(id, Philosopher::Finished);
	
}

Philosopher::Philosopher( unsigned int id, unsigned int noodles, Table &table, Printer &prt ):id(id),numberOfNoodles(noodles),table(&table),printer(&prt){}
